blog-posts/from-promises-to-observables/from-promises-to-observables.md

```diff
@@ -0,0 +1,342 @@
+
+      this.posts.push({
+        ...entry,
+        sortDate: inputDate.getTime()
+      });
+    });
+    this.posts = response.data;
+
+    // retrieve the manual entries
+    const manualEntries: any
+      = await axios.get(environment.manualEntries);
+    manualEntries.data.forEach((entry: any) => {
+      const inputDate = new Date(entry.pubDate);
+      entry.pubDate = inputDate.toLocaleDateString('en-us') + ' at ' + inputDate.toLocaleTimeString('en-us');
+      if (entry.contentSnippet.length > 200) {
+        entry.contentSnippet = entry.contentSnippet.substring(0, 200);
+      }
+
+      this.posts.push({
+        ...entry,
+        sortDate: inputDate.getTime()
+      });
+    });
+
+    // sort by date here
+    this.posts.sort((a: any, b: any) => {
+      return b.sortDate - a.sortDate;
+    });
+
+    // stop showing spinner when fetch completes
+    this.showSpinner = false;
+  }
+```
+
+Here I've created a `load` method that uses [axios](https://www.npmjs.com/package/axios) to make a GET call to retrieve the data. I also call an endpoint for an additional set of manual entries. When both complete I stop showing the spinner.
+
+## Enter RxJS
+
+So after my experiences from Angular Denver, I started thinking of ways that RxJS could improve this whole setup. I really didn't like the spinner and several second wait time, so thought this was a great opportunity to improve the site.
+
+I decided that it would help if I could break down the HTTP calls to be handled individually as streams. Then the user would see results immediately on their page, and it would update as the calls completed. Since its only a matter of seconds, this didn't make the page jump around too much and made the site feel very responsive.
+
+> I should note that typically with frontend engineering you don't want a page to jump around. In this situation it was only a matter of seconds and the way that I built the flow made it hardly noticeable. It was more important for me to show the output immediately. In a larger application this approach may need to be reworked for a better experience.
+
+I refactored the HTTP calls to be done in one endpoint. If you look in the example project, you can see this [in the functions/index.js file](https://github.com/andrewevans0102/learning-rxjs-with-angular/blob/master/functions/index.js) here.
+
+The code here is pretty straightforward, based on the "source" parameter it makes a call to to the matching RSS feed. The results are gathered from the HTTP call and returned in the output value.
+
+Now for the RxJS implementation, I wrapped each of the HTTP calls to this endpoint in a separate observable. This enabled each call to stream in at the same time, and as soon as results were found they were stored and shown on the page.
+
+```ts
+  load() {
+    const medium =
+      this.http.get(environment.reactiveEndpoint + '/medium')
```

Comment:
What about creating an higher order observable to make the call and catch the error, instead of writing the same code multiple times? Or it could be a starting point for a new article 😉

Reply:
Indeed.  I see where you were going with that.  I'm going to leave the implementation the way it is here for simplicity, as its easy to follow.  Definitely a good topic for a future post though :grin: